docker/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+func GetEnvDefault(key, defaultVal string) string {
+	val, ok := os.LookupEnv(key)
+	if !ok {
+		return defaultVal
+	}
+	return val
+}
+
+func obtainDockerPublicIP() string {
+	cmd := exec.Command("awk", "END{print $1}", "/etc/hosts")
+	var stdout bytes.Buffer
+	cmd.Stdout = &stdout
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		fmt.Print("Failed to obtain public IP:", err.Error())
+		fmt.Print(stderr.String())
+		os.Exit(1)
+	}
+
+	return strings.Trim(stdout.String(), " \n\t")
+}
+
+func getURI(port uint) string {
+	return fmt.Sprintf("%s:%d", obtainDockerPublicIP(), port)
+}
+
+const (
+	defaultAPIListenPort = 19191
+)
+
+const statusBody = "{\"jsonrpc\": \"2.0\", \"method\": \"status.Get\", \"id\": 0}"
+
+func checkInsolard() int {
+	var apiURL url.URL
+	apiURL.Scheme = "http"
+	apiURL.Host = GetEnvDefault("INSOLARD_API_LISTEN", getURI(defaultAPIListenPort))
+	apiURL.Path = "/api/rpc"
+
+	client := http.Client{
+		Transport: &http.Transport{},
+		Timeout:   60 * time.Second,
+	}
+
+	body := strings.NewReader(statusBody)
+	res, err := client.Post(apiURL.String(), "application/json", body)
+	if err != nil {
+		fmt.Printf("Failed to make HTTP request: %s", err.Error())
+		return 1
+	}
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		fmt.Printf("Failed to read body: %s", err.Error())
+		return 1
+	}
+	var out struct {
+		Result struct {
+			PulseNumber    uint32
```

Comment:
I'm almost certain that PulseNumbers are not uint32-s. Could you please recheck this, just to be sure?

Reply:
It was copy-pasted from healtcheck util

Reply:
<img width="620" alt="image" src="https://user-images.githubusercontent.com/495636/53654254-56a8b480-3c5e-11e9-8725-26f0d5660641.png">